internal/api/chisel.go

```diff
@@ -39,91 +50,134 @@ func (a *Api) StartTeleport(blocksToForward []RemoteConfig, envId string, cluste
 		utils.TerminateWithError(err)
+		return
 	}
 
 	var remotes []string
 
+	//create remotes to feed the chisel client
 	for _, remote := range blocksToForward {
-		remotes = append(
-			remotes,
-			fmt.Sprintf(
-				remote.FromHost+":"+strconv.Itoa(remote.FromPort)+":"+remote.ToHost+":"+strconv.Itoa(remote.ToPort)))
+		newRemote := fmt.Sprintf(
+			"%s:%s:%s:%s",
+			remote.FromHost,
+			strconv.Itoa(remote.FromPort),
+			remote.ToHost,
+			strconv.Itoa(remote.ToPort),
+		)
+		remotes = append(remotes, newRemote)
 	}
 
-	chiselClient, err := chclient.NewClient(&chclient.Config{
+	chiselClient, err := chisel.NewClient(&chisel.Config{
 		MaxRetryInterval: 1 * time.Second,
 		MaxRetryCount:    50,
 		Server:           fmt.Sprintf("https://%s", host.Data.Host),
 		Auth:             host.Data.Credentials,
 		Remotes:          remotes,
 		KeepAlive:        10 * time.Second,
 	})
-	defer chiselClient.Close()
 
 	if err != nil {
 		utils.TerminateWithError(err)
+		return
 	}
 
+	defer chiselClient.Close()
+
 	// Run infinite stream connection
 	go func() {
 		_, err := streamResponse.Recv()
 		if err != nil {
 			utils.TerminateWithError(err)
+			return
 		}
 	}()
 
 	chiselClient.Logger.Info = false
 
 	fmt.Println("")
 	utils.InfoMessage("Starting Tunnel")
+
+	//create remote list to display to user. check if service is
+	//teleported or port-forwarded and show msg accordingly
 	for _, remote := range blocksToForward {
-		if strings.Contains(remote.FromHost, "R:0.0.0.0") {
-			utils.InfoMessage(fmt.Sprintf("Teleporting > %s:%d => %s:%d",
-				remote.ToHost, remote.ToPort, remote.FromHost, remote.FromPort))
+		if strings.Contains(remote.FromHost, config.DefaultTeleportInterfacePort) {
+			utils.InfoMessage(fmt.Sprintf(
+				"Teleporting > %s:%d => %s:%d",
+				remote.ToHost,
+				remote.ToPort,
+				remote.FromHost,
+				remote.FromPort))
 		} else {
-			utils.InfoMessage(fmt.Sprintf("Forwarding  >  %s:%d => %s:%d",
-				remote.FromHost, remote.FromPort, remote.ToHost, remote.ToPort))
+			utils.InfoMessage(fmt.Sprintf(
+				"Forwarding  > %s:%d => %s:%d",
+				remote.FromHost,
+				remote.FromPort,
+				remote.ToHost,
+				remote.ToPort))
 		}
 	}
 	// Run chisel client in background
 	go func() {
 		err = chiselClient.Run()
 		if err != nil {
 			utils.TerminateWithError(err)
+			return
 		}
 	}()
 
 	if err != nil {
 		utils.TerminateWithError(err)
+		return
 	}
 
-	fmt.Println("")
-	utils.WarningMessage("Please start your local server at PORT => 8080")
-
-	for utils.CheckTeleportStatus(8080) {
-		utils.StartSpinner()
+	//Show start server message only if teleporting a svs
+	// For the lack of a better way, this is a temporary workaround for validating connection status
+	if isTeleport {
+		fmt.Println("")
+		//TODO: make teleport port configurable
+		utils.WarningMessage(fmt.Sprintf(
+			"Please start your local server at PORT => %d", config.DefaultClientTeleportPort))
+
+		for utils.CheckIfPortOpen(config.DefaultClientTeleportPort, false) {
```

Comment:
comment - this is english and very minor but it should be `CheckIfPortOpened` cuz u are checking if the port is opened.

Reply:
`opened` will imply that it was closed before and we are checking if it opened later on. `open` being a verb & noun implies that we are checking if the port is available/open in the first place. this is IMO. but i'll change it.

Reply:
@nadeemkhedr feel free to give ur opinion here. Cuz imo, it's `CheckIfPortIsOpened` and it does not implied that it was closed or anything. but ur english is better so give me ur .2

Reply:
Hey @sumitkolhe 
u are right actually, I had to check it https://www.quora.com/What-is-the-correct-sentence-%E2%80%9Cthe-door-is-opened%E2%80%9D-or-%E2%80%9Cthe-door-is-open%E2%80%9D-Why#:~:text=If%20someone%20asks%20if%20the,The%20door%20is%20open.%22&text=You%20have%20asked%20an%20interesting,opened%E2%80%9D%2C%20of%20the%20verb.

I am a bitch.
I love this answer particularly.

> Why do we use a past participle in “The door is closed” and “The door is shut” but an adjective in “The door is open”? Answer: I don’t know. Someone would have to do an analysis of usage over the centuries. That ain’t gonna be me. Don’t worry about it.